src/content_cao.cpp

```diff
@@ -1786,6 +1831,14 @@ void GenericCAO::processMessage(const std::string &data)
 	}
 }
 
+void GenericCAO::knockback(const v3f &direction, f32 time_knockback)
+{
+	m_knockback_timer = time_knockback;
```

Comment:
it's not hacky it's predicted, it's optimized for less workload, there is no hack there it's just a proper movement like in every MMO when you have similar animations: direction, speed + stop timer.